action/src/checkoutPublicRepository.ts

```diff
@@ -0,0 +1,26 @@
+import { ErrorStrings, GithubEventNames } from './stringEnums';
+import { context } from '@actions/github';
+import { exec } from '@actions/exec';
+
+/**
+ * Checkout the appropriate ref for the users changes using git.
+ * @returns {Promise<void>}
+ */
+export async function checkoutPublicRepository(): Promise<void> {
+  const ref = context.payload.ref;
+  const repository = context.payload.repository?.full_name;
+  try {
+    await exec('git init');
+    await exec(`git remote add origin https://github.com/${repository}.git`);
+    if (context.eventName === GithubEventNames.PULL_REQUEST) {
```

Comment:
prob can remove some code using conditional template strings for executing commands then again the is prob more readable lol : )

Reply:
ack im down with either so i can switch it up in the follow up